### DISCUSSION TITLE
Calling custom endpoint results in error ROUTE_NOT_FOUND

### DISCUSSION THREAD

**Discussion by andriusign** [directus/directus#5581] [Archive]
Custom endpoint located at directus/extensions/endpoints and looks like this:

```js
const PasswordChangeService = require('../../lib/services/pass-change-email');

module.exports = function registerEndpoint(router, { services, env, database }) {
  router.post('/password-reset', (req, res) => {
    let { email, reset_url } = req.body;

    try {
      const data = await database.where({ email }).select('email', 'status').limit(1).from('directus_users');

      if (typeof data !== 'undefined' || data.length > 0) {
        const { status } = await new PasswordChangeService().send(email, reset_url);
        console.log({ status });
        res.json({ data, status });
        return;
      }
      return res.status(404).json({ error: 'No such email found in database!' });
    } catch (e) {
      res.status(404).json({ error: e });
    }
  });
};
```

There are also other endpoints which are also not working (404).
Running in Docker. Starting with `docker-compose up --build`. Tried to rebuild containers, remove containers and then build again. Did not help. Tried also logging to Docker container and inspecting files from inside: /directus/extensions/endpoints custom endpoints are visible, files are there.

Directus _extensions_ and _uploads_ folders are mapped via docker-compose.yml:
```
    volumes:
      - ./uploads:/directus/uploads
      - ./extensions:/directus/extensions
```
    
When trying to access my endpoint I get:

```json
{"errors":[{"message":"Route /custom/user/password-reset doesn't exist.","extensions":{"code":"ROUTE_NOT_FOUND"}}]}
```

However strange thing is that the same branch is working fine on one computer, and on another two computers custom endpoint is not reachable. Same setup, same database. It used to work before locally. When deployed to dev env, it is also working fine. What might be the problem?

### Project details
Directus Version: 9.0.0-rc.64
Environment: production
OS: Linux 5.4.0-72-generic
Node: 14.16.1

**Comment by rijkvanzanten** [comment]
> However strange thing is that the same branch is working fine on one computer, and on another two computers custom endpoint is not reachable. Same setup, same database. It used to work before locally. When deployed to dev env, it is also working fine. What might be the problem?

This definitely sounds like a configuration issue to me 🤔 Are you sure the extensions folder exists on your volume? Is it mapped to the container correctly? Does the extensions path environment variable point to the correct location in the container?

**Reply by andriusign** [reply]
I have found the issue. It was related to .env file variables which were declared like:
SOME_VAR="value"

Had to change them to:
SOME_VAR=value

Applies for strings as well. So apparently it was reading `"./extensions"` value instead of `./extensions`. Now it is working fine.
